### PR TITLE
Separate code and path arguments in `RawModule`

### PR DESCRIPTION
**Pull request by niboshi**
Current API can't handle a code with a comment ending with `.cubin` at the end.

### CODE REVIEW

**Comment by niboshi** [comment]
Or maybe should we have separate interfaces for loading from code and file, instead of a single `__init__`?
E.g. `json.load` and `json.loads`.

**Comment by emcastillo** [comment]
I think the current PR solution is good enough. Adding a different interface would make the initialization process different from `ElementwiseKernel` and `ReductionKernel` and I think is better for these clases to share a similar schema.

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by emcastillo** [comment]
Needs to fix the texture tests too 😢

**Comment by emcastillo** [comment]
#2782 has been merged just now, we will need to rebase ...

**Comment by niboshi** [comment]
Fixed. PTAL

**Comment by emcastillo** [comment]
Jenkins, test this please

**Comment by leofang** [comment]
I think the change is nice as it makes the support of different kinds of valid inputs more robust. Thank you @niboshi. I am just curious about the motivation:

> Current API can't handle a code with a comment ending with `.cubin` at the end.

Who would be bothered by this? 🤣

**Comment by niboshi** [comment]
Yes, it may be unlikely, but still possible. Libraries like CuPy can't ignore such possibilities.

File: cupy/core/raw.pyx [hunk 1]
```diff
@@ -218,7 +218,8 @@ cdef class RawModule:
     similarly.
 
     Args:
-        code_or_path (str): CUDA source code or path to cubin/ptx.
+        code (str): CUDA source code. Mutually exclusive with ``path``.
+        path (str): Path to cubin/ptx. Mutually exclusive with ``code``.
```
Comment (leofang) [inline cupy/core/raw.pyx]: Sorry @niboshi, one more question: are `code` and `path` really mutually exclusive here? Based on the changes I thought a user could specify both `code` and `path` without getting an error?

Comment (leofang) [reply cupy/core/raw.pyx]: Never mind, sorry for my stupidity 😔

Comment (niboshi) [reply cupy/core/raw.pyx]: Sorry, I forgot to respond. Yes, I think they are mutually exclusive.

**Pull request merged**